commit b6b4a86dcb260ea82524a4c69b0797ce26376f34
Author: Billy Vong <billyvg@users.noreply.github.com>
Date:   Mon Jun 22 13:18:22 2020 -0700

    ref(ts): Convert `<InstallWizard>` to typescript (#19314)
    
    * Converts `<InstallWizard>` to typescript
    * Remove `setup-wizard.less` and convert to emotion
    * Move to `app/views/admin`
    * Move `app/options` to `app/views/admin`
    
    Old:
    ![image](https://user-images.githubusercontent.com/79684/84430202-2b8d6e80-abde-11ea-8d27-7bbbe63b7d15.png)
    
    New:
    ![image](https://user-images.githubusercontent.com/79684/84430146-144e8100-abde-11ea-847b-88912fb236f9.png)

diff --git a/src/sentry/static/sentry/app/utils/theme.tsx b/src/sentry/static/sentry/app/utils/theme.tsx
index ee6d2af9a5..7cf9c9d153 100644
--- a/src/sentry/static/sentry/app/utils/theme.tsx
+++ b/src/sentry/static/sentry/app/utils/theme.tsx
@@ -241,6 +241,10 @@ const theme = {
 
   // Try to keep these ordered plz
   zIndex: {
+    // Generic z-index when you hope your component is isolated and
+    // does not need to battle others for z-index priority
+    initial: 1,
+
     breadcrumbs: {
       header: 2,
       gridCellError: 1,
diff --git a/src/sentry/static/sentry/app/views/admin/adminSettings.jsx b/src/sentry/static/sentry/app/views/admin/adminSettings.jsx
index 28d24d7021..3bf7bc87db 100644
--- a/src/sentry/static/sentry/app/views/admin/adminSettings.jsx
+++ b/src/sentry/static/sentry/app/views/admin/adminSettings.jsx
@@ -3,9 +3,10 @@ import isUndefined from 'lodash/isUndefined';
 
 import AsyncView from 'app/views/asyncView';
 import {t} from 'app/locale';
-import {getOption, getOptionField} from 'app/options';
 import {ApiForm} from 'app/components/forms';
 
+import {getOption, getOptionField} from './options';
+
 const optionsAvailable = [
   'system.url-prefix',
   'system.admin-email',
diff --git a/src/sentry/static/sentry/app/views/installWizard.jsx b/src/sentry/static/sentry/app/views/admin/installWizard/index.tsx
similarity index 61%
rename from src/sentry/static/sentry/app/views/installWizard.jsx
rename to src/sentry/static/sentry/app/views/admin/installWizard/index.tsx
index 0c8047e306..c30405e274 100644
--- a/src/sentry/static/sentry/app/views/installWizard.jsx
+++ b/src/sentry/static/sentry/app/views/admin/installWizard/index.tsx
@@ -1,29 +1,27 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import DocumentTitle from 'react-document-title';
+import {css} from '@emotion/core';
+import styled from '@emotion/styled';
 
 import AsyncView from 'app/views/asyncView';
 import {t} from 'app/locale';
 import ConfigStore from 'app/stores/configStore';
 import {ApiForm} from 'app/components/forms';
-import {getOptionDefault, getOptionField, getForm} from 'app/options';
+import sentryPattern from 'app/../images/pattern/sentry-pattern.png';
+import space from 'app/styles/space';
+import Alert from 'app/components/alert';
+import {IconWarning} from 'app/icons';
 
-export default class InstallWizard extends AsyncView {
-  static propTypes = {
-    onConfigured: PropTypes.func.isRequired,
-  };
+import {getOptionDefault, getOptionField, getForm} from '../options';
 
-  UNSAFE_componentWillMount() {
-    super.UNSAFE_componentWillMount();
-    document.body.classList.add('install-wizard');
-  }
+type Props = {
+  onConfigured: () => void;
+} & AsyncView['props'];
 
-  componentWillUnmount() {
-    super.componentWillUnmount();
-    document.body.classList.remove('install-wizard');
-  }
+type State = {} & AsyncView['state'];
 
-  getEndpoints() {
+export default class InstallWizard extends AsyncView<Props, State> {
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
     return [['data', '/internal/options/?query=is:required']];
   }
 
@@ -92,32 +90,31 @@ export default class InstallWizard extends AsyncView {
     const version = ConfigStore.get('version');
     return (
       <DocumentTitle title={this.getTitle()}>
-        <div className="app">
-          <div className="pattern" />
-          <div className="setup-wizard">
-            <h1>
+        <Wrapper>
+          <Pattern />
+          <SetupWizard>
+            <Heading>
               <span>{t('Welcome to Sentry')}</span>
-              <small>{version.current}</small>
-            </h1>
+              <Version>{version.current}</Version>
+            </Heading>
             {this.state.loading
               ? this.renderLoading()
               : this.state.error
-              ? this.renderError(new Error('Unable to load all required endpoints'))
+              ? this.renderError()
               : this.renderBody()}
-          </div>
-        </div>
+          </SetupWizard>
+        </Wrapper>
       </DocumentTitle>
     );
   }
 
   renderError() {
     return (
-      <div className="loading-error">
-        <span className="icon-exclamation" />
+      <Alert type="error" icon={<IconWarning />}>
         {t(
           'We were unable to load the required configuration from the Sentry server. Please take a look at the service logs.'
         )}
-      </div>
+      </Alert>
     );
   }
 
@@ -137,3 +134,60 @@ export default class InstallWizard extends AsyncView {
     );
   }
 }
+
+const Wrapper = styled('div')`
+  display: flex;
+  justify-content: center;
+`;
+
+const fixedStyle = css`
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+`;
+
+const Pattern = styled('div')`
+  &::before {
+    ${fixedStyle}
+    content: '';
+    background-image: linear-gradient(
+      to right,
+      ${p => p.theme.purple300} 0%,
+      ${p => p.theme.purple400} 100%
+    );
+    background-repeat: repeat-y;
+  }
+
+  &::after {
+    ${fixedStyle}
+    content: '';
+    background: url(${sentryPattern});
+    background-size: 400px;
+    opacity: 0.8;
+  }
+`;
+
+const Heading = styled('h1')`
+  display: grid;
+  grid-gap: ${space(1)};
+  justify-content: space-between;
+  grid-auto-flow: column;
+  line-height: 36px;
+`;
+
+const Version = styled('small')`
+  font-size: ${p => p.theme.fontSizeExtraLarge};
+  line-height: inherit;
+`;
+
+const SetupWizard = styled('div')`
+  background: ${p => p.theme.white};
+  border-radius: ${p => p.theme.borderRadius};
+  box-shadow: ${p => p.theme.dropShadowHeavy};
+  margin-top: 40px;
+  padding: 40px 40px 20px;
+  width: 600px;
+  z-index: ${p => p.theme.zIndex.initial};
+`;
diff --git a/src/sentry/static/sentry/app/options.jsx b/src/sentry/static/sentry/app/views/admin/options.jsx
similarity index 100%
rename from src/sentry/static/sentry/app/options.jsx
rename to src/sentry/static/sentry/app/views/admin/options.jsx
diff --git a/src/sentry/static/sentry/app/views/app/index.tsx b/src/sentry/static/sentry/app/views/app/index.tsx
index 250662df75..53d909da80 100644
--- a/src/sentry/static/sentry/app/views/app/index.tsx
+++ b/src/sentry/static/sentry/app/views/app/index.tsx
@@ -234,7 +234,7 @@ class App extends React.Component<Props, State> {
 
     if (needsUpgrade) {
       const InstallWizard = React.lazy(() =>
-        import(/* webpackChunkName: "InstallWizard" */ 'app/views/installWizard')
+        import(/* webpackChunkName: "InstallWizard" */ 'app/views/admin/installWizard')
       );
 
       return (
diff --git a/src/sentry/static/sentry/less/sentry.less b/src/sentry/static/sentry/less/sentry.less
index 32b3bdec52..a86341fc1f 100644
--- a/src/sentry/static/sentry/less/sentry.less
+++ b/src/sentry/static/sentry/less/sentry.less
@@ -32,7 +32,6 @@
 @import url('./organization.less');
 @import url('./project-settings.less');
 @import url('./releases.less');
-@import url('./setup-wizard.less');
 @import url('./spacing.less');
 @import url('./stream.less');
 @import url('./group-similar.less');
diff --git a/src/sentry/static/sentry/less/setup-wizard.less b/src/sentry/static/sentry/less/setup-wizard.less
deleted file mode 100644
index 33cd39c0fa..0000000000
--- a/src/sentry/static/sentry/less/setup-wizard.less
+++ /dev/null
@@ -1,69 +0,0 @@
-/**
-* body.install-wizard
-* ============================================================================
-*/
-
-body.install-wizard {
-  #gradient > .horizontal(#9f73c3, #6875be);
-
-  background-repeat: repeat-y;
-
-  .pattern {
-    position: fixed;
-    top: 0;
-    right: 0;
-    bottom: 0;
-    left: 0;
-    background: url(../images/pattern/sentry-pattern.png);
-    background-size: 400px;
-    opacity: 0.8;
-  }
-}
-
-.setup-wizard {
-  .clearfix;
-  background: #fff;
-  width: 600px;
-  padding: 40px 40px 20px;
-  border-radius: 4px;
-  box-shadow: 0 5px 30px rgba(0, 0, 0, 0.3);
-  position: absolute;
-  top: 50px;
-  left: 50%;
-  margin-left: -300px;
-  margin-bottom: 50px;
-
-  h1 {
-    .clearfix;
-    line-height: 36px;
-
-    > span {
-      display: block;
-      float: left;
-    }
-    > small {
-      float: right;
-      display: block;
-      font-size: 18px;
-      line-height: inherit;
-    }
-  }
-
-  .loading {
-    margin: 0 0 20px;
-  }
-
-  .loading-error {
-    .clearfix;
-    margin: 0 0 20px;
-
-    .icon:before {
-      float: left;
-      display: block;
-      font-size: 50px;
-      margin-right: 10px;
-      color: @red;
-      content: '\e615';
-    }
-  }
-}
diff --git a/tests/js/spec/components/events/interfaces/breadcrumbs/__snapshots__/filter.spec.tsx.snap b/tests/js/spec/components/events/interfaces/breadcrumbs/__snapshots__/filter.spec.tsx.snap
index 3f09c05a49..d7838c85c4 100644
--- a/tests/js/spec/components/events/interfaces/breadcrumbs/__snapshots__/filter.spec.tsx.snap
+++ b/tests/js/spec/components/events/interfaces/breadcrumbs/__snapshots__/filter.spec.tsx.snap
@@ -716,6 +716,7 @@ exports[`Filter default render 1`] = `
                       "globalSelectionHeader": 1009,
                       "header": 1000,
                       "hovercard": 10002,
+                      "initial": 1,
                       "modal": 10000,
                       "orgAndUserMenu": 1030,
                       "sentryErrorEmbed": 1090,
diff --git a/tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap b/tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap
deleted file mode 100644
index 9eb8056d32..0000000000
--- a/tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap
+++ /dev/null
@@ -1,558 +0,0 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
-
-exports[`InstallWizard renders 1`] = `
-<InstallWizard
-  onConfigured={[MockFunction]}
->
-  <SideEffect(DocumentTitle)
-    title="Setup Sentry"
-  >
-    <DocumentTitle
-      title="Setup Sentry"
-    >
-      <div
-        className="app"
-      >
-        <div
-          className="pattern"
-        />
-        <div
-          className="setup-wizard"
-        >
-          <h1>
-            <span>
-              Welcome to Sentry
-            </span>
-            <small>
-              1.33.7
-            </small>
-          </h1>
-          <ApiForm
-            apiEndpoint="/internal/options/?query=is:required"
-            apiMethod="PUT"
-            cancelLabel="Cancel"
-            className="form-stacked"
-            errorMessage="Unable to save your changes. Please ensure all fields are valid and try again."
-            footerClass="form-actions align-right"
-            hideErrors={false}
-            initialData={Object {}}
-            onSubmitSuccess={[MockFunction]}
-            requireChanges={false}
-            resetOnError={false}
-            submitDisabled={false}
-            submitErrorMessage="There was an error saving your changes."
-            submitLabel="Continue"
-            submitLoadingMessage="Saving changes.."
-          >
-            <StyledForm
-              className="form-stacked"
-              onSubmit={[Function]}
-            >
-              <form
-                className="form-stacked css-1z07nlv-StyledForm e18zovyk0"
-                onSubmit={[Function]}
-              >
-                <p>
-                  Complete setup by filling out the required configuration.
-                </p>
-                <fieldset
-                  key="system"
-                >
-                  <EmailField
-                    allowEmpty={false}
-                    component={[Function]}
-                    default=""
-                    disabled={false}
-                    disabledReason={null}
-                    help="The technical contact for this Sentry installation."
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="system.admin-email"
-                    label="Admin Email"
-                    name="system.admin-email"
-                    placeholder="admin@example.com"
-                    required={true}
-                  >
-                    <div
-                      className="control-group required"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                          htmlFor="id-system.admin-email"
-                        >
-                          Admin Email
-                        </label>
-                        <input
-                          className="form-control"
-                          disabled={false}
-                          id="id-system.admin-email"
-                          name="system.admin-email"
-                          onChange={[Function]}
-                          placeholder="admin@example.com"
-                          required={true}
-                          type="email"
-                          value=""
-                        />
-                        <p
-                          className="help-block"
-                        >
-                          The technical contact for this Sentry installation.
-                        </p>
-                      </div>
-                    </div>
-                  </EmailField>
-                </fieldset>
-                <fieldset
-                  key="mail"
-                >
-                  <legend>
-                    Outbound email
-                  </legend>
-                  <EmailField
-                    allowEmpty={false}
-                    component={[Function]}
-                    default="root@localhost"
-                    defaultValue="sentry@localhost"
-                    disabled={false}
-                    disabledReason={null}
-                    help="Email address to be used in From for all outbound email."
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="mail.from"
-                    label="Email From"
-                    name="mail.from"
-                    required={true}
-                  >
-                    <div
-                      className="control-group required"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                          htmlFor="id-mail.from"
-                        >
-                          Email From
-                        </label>
-                        <input
-                          className="form-control"
-                          disabled={false}
-                          id="id-mail.from"
-                          name="mail.from"
-                          onChange={[Function]}
-                          required={true}
-                          type="email"
-                          value="sentry@localhost"
-                        />
-                        <p
-                          className="help-block"
-                        >
-                          Email address to be used in From for all outbound email.
-                        </p>
-                      </div>
-                    </div>
-                  </EmailField>
-                  <TextField
-                    allowEmpty={false}
-                    default="localhost"
-                    defaultValue="localhost"
-                    disabled={false}
-                    disabledReason={null}
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="mail.host"
-                    label="SMTP Host"
-                    name="mail.host"
-                    placeholder="localhost"
-                    required={true}
-                  >
-                    <div
-                      className="control-group required"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                          htmlFor="id-mail.host"
-                        >
-                          SMTP Host
-                        </label>
-                        <input
-                          className="form-control"
-                          disabled={false}
-                          id="id-mail.host"
-                          name="mail.host"
-                          onChange={[Function]}
-                          placeholder="localhost"
-                          required={true}
-                          type="text"
-                          value="localhost"
-                        />
-                      </div>
-                    </div>
-                  </TextField>
-                  <TextField
-                    allowEmpty={false}
-                    default={25}
-                    defaultValue="25"
-                    disabled={false}
-                    disabledReason={null}
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="mail.port"
-                    label="SMTP Port"
-                    name="mail.port"
-                    placeholder="25"
-                    required={true}
-                  >
-                    <div
-                      className="control-group required"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                          htmlFor="id-mail.port"
-                        >
-                          SMTP Port
-                        </label>
-                        <input
-                          className="form-control"
-                          disabled={false}
-                          id="id-mail.port"
-                          name="mail.port"
-                          onChange={[Function]}
-                          placeholder="25"
-                          required={true}
-                          type="text"
-                          value="25"
-                        />
-                      </div>
-                    </div>
-                  </TextField>
-                  <TextField
-                    allowEmpty={true}
-                    default=""
-                    defaultValue=""
-                    disabled={false}
-                    disabledReason={null}
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="mail.username"
-                    label="SMTP Username"
-                    name="mail.username"
-                    required={false}
-                  >
-                    <div
-                      className="control-group"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                          htmlFor="id-mail.username"
-                        >
-                          SMTP Username
-                        </label>
-                        <input
-                          className="form-control"
-                          disabled={false}
-                          id="id-mail.username"
-                          name="mail.username"
-                          onChange={[Function]}
-                          required={false}
-                          type="text"
-                          value=""
-                        />
-                      </div>
-                    </div>
-                  </TextField>
-                  <TextField
-                    allowEmpty={true}
-                    default=""
-                    defaultValue=""
-                    disabled={false}
-                    disabledReason={null}
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="mail.password"
-                    label="SMTP Password"
-                    name="mail.password"
-                    required={false}
-                  >
-                    <div
-                      className="control-group"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                          htmlFor="id-mail.password"
-                        >
-                          SMTP Password
-                        </label>
-                        <input
-                          className="form-control"
-                          disabled={false}
-                          id="id-mail.password"
-                          name="mail.password"
-                          onChange={[Function]}
-                          required={false}
-                          type="text"
-                          value=""
-                        />
-                      </div>
-                    </div>
-                  </TextField>
-                  <BooleanField
-                    allowEmpty={true}
-                    component={[Function]}
-                    default={false}
-                    defaultValue={false}
-                    disabled={false}
-                    disabledReason={null}
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="mail.use-tls"
-                    label="Use TLS?"
-                    name="mail.use-tls"
-                    required={false}
-                  >
-                    <div
-                      className="control-group checkbox"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                        >
-                          <input
-                            checked={false}
-                            disabled={false}
-                            id="id-mail.use-tls"
-                            onChange={[Function]}
-                            type="checkbox"
-                          />
-                          Use TLS?
-                        </label>
-                      </div>
-                    </div>
-                  </BooleanField>
-                </fieldset>
-                <fieldset
-                  key="auth"
-                >
-                  <legend>
-                    Authentication
-                  </legend>
-                  <BooleanField
-                    allowEmpty={true}
-                    component={[Function]}
-                    default={false}
-                    defaultValue={false}
-                    disabled={false}
-                    disabledReason={null}
-                    help="Allow anyone to create an account and access this Sentry installation."
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="auth.allow-registration"
-                    label="Allow Registration"
-                    name="auth.allow-registration"
-                    required={false}
-                  >
-                    <div
-                      className="control-group checkbox"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                        >
-                          <input
-                            checked={false}
-                            disabled={false}
-                            id="id-auth.allow-registration"
-                            onChange={[Function]}
-                            type="checkbox"
-                          />
-                          Allow Registration
-                        </label>
-                        <p
-                          className="help-block"
-                        >
-                          Allow anyone to create an account and access this Sentry installation.
-                        </p>
-                      </div>
-                    </div>
-                  </BooleanField>
-                </fieldset>
-                <fieldset
-                  key="beacon"
-                >
-                  <legend>
-                    Beacon
-                  </legend>
-                  <RadioBooleanField
-                    allowEmpty={true}
-                    component={[Function]}
-                    default={false}
-                    disabled={false}
-                    disabledReason={null}
-                    help={
-                      <React.Fragment>
-                        <span>
-                          <span>
-                            If enabled, any stats reported to sentry.io will exclude identifying information (such as your administrative email address). By anonymizing your installation the Sentry team will be unable to contact you about security updates. For more information on what data is sent to Sentry, see the 
-                          </span>
-                          <a
-                            href="https://docs.sentry.io/server/beacon/"
-                          >
-                            <span>
-                              documentation
-                            </span>
-                          </a>
-                          <span>
-                            .
-                          </span>
-                        </span>
-                      </React.Fragment>
-                    }
-                    hideErrorMessage={false}
-                    isSet={true}
-                    key="beacon.anonymous"
-                    label="Usage Statistics"
-                    name="beacon.anonymous"
-                    noLabel="Send my contact information along with usage statistics"
-                    required={false}
-                    yesFirst={false}
-                    yesLabel="Please keep my usage information anonymous"
-                  >
-                    <div
-                      className="control-group"
-                    >
-                      <div
-                        className="controls"
-                      >
-                        <label
-                          className="control-label"
-                          htmlFor="id-beacon.anonymous"
-                        >
-                          Usage Statistics
-                        </label>
-                        <p
-                          className="help-block"
-                        >
-                          <span
-                            key="5"
-                          >
-                            <span
-                              key="0"
-                            >
-                              If enabled, any stats reported to sentry.io will exclude identifying information (such as your administrative email address). By anonymizing your installation the Sentry team will be unable to contact you about security updates. For more information on what data is sent to Sentry, see the 
-                            </span>
-                            <a
-                              href="https://docs.sentry.io/server/beacon/"
-                              key="2"
-                            >
-                              <span
-                                key="1"
-                              >
-                                documentation
-                              </span>
-                            </a>
-                            <span
-                              key="3"
-                            >
-                              .
-                            </span>
-                          </span>
-                        </p>
-                        <div
-                          className="control-group radio-boolean"
-                        >
-                          <div
-                            className="radio"
-                            key="no"
-                          >
-                            <label
-                              style={
-                                Object {
-                                  "fontWeight": "normal",
-                                }
-                              }
-                            >
-                              <input
-                                checked={false}
-                                disabled={false}
-                                name="beacon.anonymous"
-                                onChange={[Function]}
-                                type="radio"
-                                value="false"
-                              />
-                               
-                              Send my contact information along with usage statistics
-                            </label>
-                          </div>
-                          <div
-                            className="radio"
-                            key="yes"
-                          >
-                            <label
-                              style={
-                                Object {
-                                  "fontWeight": "normal",
-                                }
-                              }
-                            >
-                              <input
-                                checked={false}
-                                disabled={false}
-                                name="beacon.anonymous"
-                                onChange={[Function]}
-                                type="radio"
-                                value="true"
-                              />
-                               
-                              Please keep my usage information anonymous
-                            </label>
-                          </div>
-                        </div>
-                      </div>
-                    </div>
-                  </RadioBooleanField>
-                </fieldset>
-                <div
-                  className="form-actions align-right"
-                  style={
-                    Object {
-                      "marginTop": 25,
-                    }
-                  }
-                >
-                  <button
-                    className="btn btn-primary"
-                    disabled={false}
-                    type="submit"
-                  >
-                    Continue
-                  </button>
-                </div>
-              </form>
-            </StyledForm>
-          </ApiForm>
-        </div>
-      </div>
-    </DocumentTitle>
-  </SideEffect(DocumentTitle)>
-</InstallWizard>
-`;
diff --git a/tests/js/spec/views/admin/__snapshots__/installWizard.spec.jsx.snap b/tests/js/spec/views/admin/__snapshots__/installWizard.spec.jsx.snap
new file mode 100644
index 0000000000..5f10b50b1e
--- /dev/null
+++ b/tests/js/spec/views/admin/__snapshots__/installWizard.spec.jsx.snap
@@ -0,0 +1,572 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`InstallWizard renders 1`] = `
+<InstallWizard
+  onConfigured={[MockFunction]}
+>
+  <SideEffect(DocumentTitle)
+    title="Setup Sentry"
+  >
+    <DocumentTitle
+      title="Setup Sentry"
+    >
+      <Wrapper>
+        <div
+          className="css-vttdei-Wrapper e1c7mu5x0"
+        >
+          <Pattern>
+            <div
+              className="css-xf94ab-Pattern-fixedStyle-fixedStyle e1c7mu5x1"
+            />
+          </Pattern>
+          <SetupWizard>
+            <div
+              className="css-do021x-SetupWizard e1c7mu5x4"
+            >
+              <Heading>
+                <h1
+                  className="css-hclgk-Heading e1c7mu5x2"
+                >
+                  <span>
+                    Welcome to Sentry
+                  </span>
+                  <Version>
+                    <small
+                      className="css-qqvcsv-Version e1c7mu5x3"
+                    >
+                      1.33.7
+                    </small>
+                  </Version>
+                </h1>
+              </Heading>
+              <ApiForm
+                apiEndpoint="/internal/options/?query=is:required"
+                apiMethod="PUT"
+                cancelLabel="Cancel"
+                className="form-stacked"
+                errorMessage="Unable to save your changes. Please ensure all fields are valid and try again."
+                footerClass="form-actions align-right"
+                hideErrors={false}
+                initialData={Object {}}
+                onSubmitSuccess={[MockFunction]}
+                requireChanges={false}
+                resetOnError={false}
+                submitDisabled={false}
+                submitErrorMessage="There was an error saving your changes."
+                submitLabel="Continue"
+                submitLoadingMessage="Saving changes.."
+              >
+                <StyledForm
+                  className="form-stacked"
+                  onSubmit={[Function]}
+                >
+                  <form
+                    className="form-stacked css-1z07nlv-StyledForm e18zovyk0"
+                    onSubmit={[Function]}
+                  >
+                    <p>
+                      Complete setup by filling out the required configuration.
+                    </p>
+                    <fieldset
+                      key="system"
+                    >
+                      <EmailField
+                        allowEmpty={false}
+                        component={[Function]}
+                        default=""
+                        disabled={false}
+                        disabledReason={null}
+                        help="The technical contact for this Sentry installation."
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="system.admin-email"
+                        label="Admin Email"
+                        name="system.admin-email"
+                        placeholder="admin@example.com"
+                        required={true}
+                      >
+                        <div
+                          className="control-group required"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                              htmlFor="id-system.admin-email"
+                            >
+                              Admin Email
+                            </label>
+                            <input
+                              className="form-control"
+                              disabled={false}
+                              id="id-system.admin-email"
+                              name="system.admin-email"
+                              onChange={[Function]}
+                              placeholder="admin@example.com"
+                              required={true}
+                              type="email"
+                              value=""
+                            />
+                            <p
+                              className="help-block"
+                            >
+                              The technical contact for this Sentry installation.
+                            </p>
+                          </div>
+                        </div>
+                      </EmailField>
+                    </fieldset>
+                    <fieldset
+                      key="mail"
+                    >
+                      <legend>
+                        Outbound email
+                      </legend>
+                      <EmailField
+                        allowEmpty={false}
+                        component={[Function]}
+                        default="root@localhost"
+                        defaultValue="sentry@localhost"
+                        disabled={false}
+                        disabledReason={null}
+                        help="Email address to be used in From for all outbound email."
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="mail.from"
+                        label="Email From"
+                        name="mail.from"
+                        required={true}
+                      >
+                        <div
+                          className="control-group required"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                              htmlFor="id-mail.from"
+                            >
+                              Email From
+                            </label>
+                            <input
+                              className="form-control"
+                              disabled={false}
+                              id="id-mail.from"
+                              name="mail.from"
+                              onChange={[Function]}
+                              required={true}
+                              type="email"
+                              value="sentry@localhost"
+                            />
+                            <p
+                              className="help-block"
+                            >
+                              Email address to be used in From for all outbound email.
+                            </p>
+                          </div>
+                        </div>
+                      </EmailField>
+                      <TextField
+                        allowEmpty={false}
+                        default="localhost"
+                        defaultValue="localhost"
+                        disabled={false}
+                        disabledReason={null}
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="mail.host"
+                        label="SMTP Host"
+                        name="mail.host"
+                        placeholder="localhost"
+                        required={true}
+                      >
+                        <div
+                          className="control-group required"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                              htmlFor="id-mail.host"
+                            >
+                              SMTP Host
+                            </label>
+                            <input
+                              className="form-control"
+                              disabled={false}
+                              id="id-mail.host"
+                              name="mail.host"
+                              onChange={[Function]}
+                              placeholder="localhost"
+                              required={true}
+                              type="text"
+                              value="localhost"
+                            />
+                          </div>
+                        </div>
+                      </TextField>
+                      <TextField
+                        allowEmpty={false}
+                        default={25}
+                        defaultValue="25"
+                        disabled={false}
+                        disabledReason={null}
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="mail.port"
+                        label="SMTP Port"
+                        name="mail.port"
+                        placeholder="25"
+                        required={true}
+                      >
+                        <div
+                          className="control-group required"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                              htmlFor="id-mail.port"
+                            >
+                              SMTP Port
+                            </label>
+                            <input
+                              className="form-control"
+                              disabled={false}
+                              id="id-mail.port"
+                              name="mail.port"
+                              onChange={[Function]}
+                              placeholder="25"
+                              required={true}
+                              type="text"
+                              value="25"
+                            />
+                          </div>
+                        </div>
+                      </TextField>
+                      <TextField
+                        allowEmpty={true}
+                        default=""
+                        defaultValue=""
+                        disabled={false}
+                        disabledReason={null}
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="mail.username"
+                        label="SMTP Username"
+                        name="mail.username"
+                        required={false}
+                      >
+                        <div
+                          className="control-group"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                              htmlFor="id-mail.username"
+                            >
+                              SMTP Username
+                            </label>
+                            <input
+                              className="form-control"
+                              disabled={false}
+                              id="id-mail.username"
+                              name="mail.username"
+                              onChange={[Function]}
+                              required={false}
+                              type="text"
+                              value=""
+                            />
+                          </div>
+                        </div>
+                      </TextField>
+                      <TextField
+                        allowEmpty={true}
+                        default=""
+                        defaultValue=""
+                        disabled={false}
+                        disabledReason={null}
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="mail.password"
+                        label="SMTP Password"
+                        name="mail.password"
+                        required={false}
+                      >
+                        <div
+                          className="control-group"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                              htmlFor="id-mail.password"
+                            >
+                              SMTP Password
+                            </label>
+                            <input
+                              className="form-control"
+                              disabled={false}
+                              id="id-mail.password"
+                              name="mail.password"
+                              onChange={[Function]}
+                              required={false}
+                              type="text"
+                              value=""
+                            />
+                          </div>
+                        </div>
+                      </TextField>
+                      <BooleanField
+                        allowEmpty={true}
+                        component={[Function]}
+                        default={false}
+                        defaultValue={false}
+                        disabled={false}
+                        disabledReason={null}
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="mail.use-tls"
+                        label="Use TLS?"
+                        name="mail.use-tls"
+                        required={false}
+                      >
+                        <div
+                          className="control-group checkbox"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                            >
+                              <input
+                                checked={false}
+                                disabled={false}
+                                id="id-mail.use-tls"
+                                onChange={[Function]}
+                                type="checkbox"
+                              />
+                              Use TLS?
+                            </label>
+                          </div>
+                        </div>
+                      </BooleanField>
+                    </fieldset>
+                    <fieldset
+                      key="auth"
+                    >
+                      <legend>
+                        Authentication
+                      </legend>
+                      <BooleanField
+                        allowEmpty={true}
+                        component={[Function]}
+                        default={false}
+                        defaultValue={false}
+                        disabled={false}
+                        disabledReason={null}
+                        help="Allow anyone to create an account and access this Sentry installation."
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="auth.allow-registration"
+                        label="Allow Registration"
+                        name="auth.allow-registration"
+                        required={false}
+                      >
+                        <div
+                          className="control-group checkbox"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                            >
+                              <input
+                                checked={false}
+                                disabled={false}
+                                id="id-auth.allow-registration"
+                                onChange={[Function]}
+                                type="checkbox"
+                              />
+                              Allow Registration
+                            </label>
+                            <p
+                              className="help-block"
+                            >
+                              Allow anyone to create an account and access this Sentry installation.
+                            </p>
+                          </div>
+                        </div>
+                      </BooleanField>
+                    </fieldset>
+                    <fieldset
+                      key="beacon"
+                    >
+                      <legend>
+                        Beacon
+                      </legend>
+                      <RadioBooleanField
+                        allowEmpty={true}
+                        component={[Function]}
+                        default={false}
+                        disabled={false}
+                        disabledReason={null}
+                        help={
+                          <React.Fragment>
+                            <span>
+                              <span>
+                                If enabled, any stats reported to sentry.io will exclude identifying information (such as your administrative email address). By anonymizing your installation the Sentry team will be unable to contact you about security updates. For more information on what data is sent to Sentry, see the 
+                              </span>
+                              <a
+                                href="https://docs.sentry.io/server/beacon/"
+                              >
+                                <span>
+                                  documentation
+                                </span>
+                              </a>
+                              <span>
+                                .
+                              </span>
+                            </span>
+                          </React.Fragment>
+                        }
+                        hideErrorMessage={false}
+                        isSet={true}
+                        key="beacon.anonymous"
+                        label="Usage Statistics"
+                        name="beacon.anonymous"
+                        noLabel="Send my contact information along with usage statistics"
+                        required={false}
+                        yesFirst={false}
+                        yesLabel="Please keep my usage information anonymous"
+                      >
+                        <div
+                          className="control-group"
+                        >
+                          <div
+                            className="controls"
+                          >
+                            <label
+                              className="control-label"
+                              htmlFor="id-beacon.anonymous"
+                            >
+                              Usage Statistics
+                            </label>
+                            <p
+                              className="help-block"
+                            >
+                              <span
+                                key="5"
+                              >
+                                <span
+                                  key="0"
+                                >
+                                  If enabled, any stats reported to sentry.io will exclude identifying information (such as your administrative email address). By anonymizing your installation the Sentry team will be unable to contact you about security updates. For more information on what data is sent to Sentry, see the 
+                                </span>
+                                <a
+                                  href="https://docs.sentry.io/server/beacon/"
+                                  key="2"
+                                >
+                                  <span
+                                    key="1"
+                                  >
+                                    documentation
+                                  </span>
+                                </a>
+                                <span
+                                  key="3"
+                                >
+                                  .
+                                </span>
+                              </span>
+                            </p>
+                            <div
+                              className="control-group radio-boolean"
+                            >
+                              <div
+                                className="radio"
+                                key="no"
+                              >
+                                <label
+                                  style={
+                                    Object {
+                                      "fontWeight": "normal",
+                                    }
+                                  }
+                                >
+                                  <input
+                                    checked={false}
+                                    disabled={false}
+                                    name="beacon.anonymous"
+                                    onChange={[Function]}
+                                    type="radio"
+                                    value="false"
+                                  />
+                                   
+                                  Send my contact information along with usage statistics
+                                </label>
+                              </div>
+                              <div
+                                className="radio"
+                                key="yes"
+                              >
+                                <label
+                                  style={
+                                    Object {
+                                      "fontWeight": "normal",
+                                    }
+                                  }
+                                >
+                                  <input
+                                    checked={false}
+                                    disabled={false}
+                                    name="beacon.anonymous"
+                                    onChange={[Function]}
+                                    type="radio"
+                                    value="true"
+                                  />
+                                   
+                                  Please keep my usage information anonymous
+                                </label>
+                              </div>
+                            </div>
+                          </div>
+                        </div>
+                      </RadioBooleanField>
+                    </fieldset>
+                    <div
+                      className="form-actions align-right"
+                      style={
+                        Object {
+                          "marginTop": 25,
+                        }
+                      }
+                    >
+                      <button
+                        className="btn btn-primary"
+                        disabled={false}
+                        type="submit"
+                      >
+                        Continue
+                      </button>
+                    </div>
+                  </form>
+                </StyledForm>
+              </ApiForm>
+            </div>
+          </SetupWizard>
+        </div>
+      </Wrapper>
+    </DocumentTitle>
+  </SideEffect(DocumentTitle)>
+</InstallWizard>
+`;
diff --git a/tests/js/spec/views/installWizard.spec.jsx b/tests/js/spec/views/admin/installWizard.spec.jsx
similarity index 85%
rename from tests/js/spec/views/installWizard.spec.jsx
rename to tests/js/spec/views/admin/installWizard.spec.jsx
index aaf2c7156b..ce5d5491f1 100644
--- a/tests/js/spec/views/installWizard.spec.jsx
+++ b/tests/js/spec/views/admin/installWizard.spec.jsx
@@ -1,9 +1,9 @@
 import React from 'react';
 
-import {mount} from 'sentry-test/enzyme';
+import {mountWithTheme} from 'sentry-test/enzyme';
 
 import ConfigStore from 'app/stores/configStore';
-import InstallWizard from 'app/views/installWizard';
+import InstallWizard from 'app/views/admin/installWizard';
 
 jest.mock('app/stores/configStore', () => ({
   get: jest.fn(),
@@ -28,7 +28,7 @@ describe('InstallWizard', function() {
   beforeEach(function() {});
 
   it('renders', function() {
-    const wrapper = mount(<InstallWizard onConfigured={jest.fn()} />);
+    const wrapper = mountWithTheme(<InstallWizard onConfigured={jest.fn()} />);
     expect(wrapper).toMatchSnapshot();
   });
 
@@ -49,7 +49,7 @@ describe('InstallWizard', function() {
         },
       }),
     });
-    const wrapper = mount(<InstallWizard onConfigured={jest.fn()} />);
+    const wrapper = mountWithTheme(<InstallWizard onConfigured={jest.fn()} />);
 
     expect(
       wrapper.find('input[name="beacon.anonymous"][value="false"]').prop('checked')
@@ -77,7 +77,7 @@ describe('InstallWizard', function() {
         },
       }),
     });
-    const wrapper = mount(<InstallWizard onConfigured={jest.fn()} />);
+    const wrapper = mountWithTheme(<InstallWizard onConfigured={jest.fn()} />);
 
     expect(
       wrapper.find('input[name="beacon.anonymous"][value="false"]').prop('checked')
